### ISSUE TITLE
Load user translation files to extend core translations

### ISSUE THREAD

**Issue by sbani** [bolt/bolt#3553]
If you use bolt/composer-install, the translation files are placed in vendor dir, but should be placed in /app/ where the configs are placed.
An even better way is to let the user overwrite the current translations with his files in his app dir.

Else, the an composer update will overwrite the translation files

**Comment by rarila** [comment]
Having those in config/ is on my list since at least 2.0. The place where they are now should as dist files then. Problem is there's much more code that needs some love than spare time :-(

**Comment by rossriley** [comment]
There's a fairly simple workaround for this while we're waiting for native support, if you have a service provider in your app, (or don't mind adding to the bootstrap file some similar code you can do this..)

    public function boot(Application $app)
    {
        $path = $app['resources']->getPath('root/app/resources/translations/en_GB');
        $app['translator']->addResource('yml', $path.'/overrides.en.yml', 'en_GB', 'messages');
    }

Then you can override anything in the messages.yml file and it will take precedence over the default bolt translations.

Something like this, for instance in `app/resources/translations/en_GB/overrides.yml`

     page:
        dashboard:
            title: "Control Centre"

**Comment by bobdenotter** [comment]
> "Translation files are placed wrong"

Just stating for the record.. Because you would like them in a different location, that doesn't mean it's currently _wrong_.

**Comment by fabschurt** [comment]
Nice trick @rossriley, thanks! So far, I was overriding `ResourceManager::getPath()` in a subclass, which was far from ideal:

```php
public function getPath($name)
{
    if (preg_match('#^app/resources/translations/[a-z]{2}(?:_[A-Z]{2})?$#', $name, $matches)) {
        return parent::getPath("root/{$matches[0]}");
    }

    return parent::getPath($name);
}
```

Anyway, the problem here is not that translation files are placed _wrong_ or anything, it's just that Bolt by default doesn't load other translation resources than the core resources. :+1: for the upcoming native support, even though the workaround is clean and easy.

**Comment by fabschurt** [comment]
Would #3800 do the trick?

**Comment by SahAssar** [comment]
Closing since this was fixed in #3800, no?

If that is not the case please feel free to reopen! :)

**Comment by sbani** [comment]
Thanks @SahAssar and @fabschurt. Very busy these times I'm sorry!
